packages/expo-cli/src/commands/upload/IOSUploader.ts

```diff
@@ -110,16 +121,22 @@ export default class IOSUploader extends BaseUploader {
     };
   }
 
-  async _getAppleTeamId() {
+  async _getAppleTeamId(): Promise<string | undefined> {
     const credentialMetadata = await Credentials.getCredentialMetadataAsync(this.projectDir, 'ios');
-    const { teamId } = await Credentials.getCredentialsForPlatform(credentialMetadata);
+    const credential = await Credentials.getCredentialsForPlatform(credentialMetadata);
 
-    return teamId;
+    if (credential) {
+      return credential.teamId;
+    }
+    return;
   }
 
-  async _getAppleIdCredentials() {
+  async _getAppleIdCredentials(): Promise<Pick<IosPlatformOptions, 'appleId' | 'appleIdPassword'>> {
     const appleCredsKeys = ['appleId', 'appleIdPassword'];
-    const result = pick(this.options, appleCredsKeys);
+    const result: Partial<Pick<IosPlatformOptions, 'appleId' | 'appleIdPassword'>> = pick(
+      this.options,
+      appleCredsKeys
+    );
```

Comment:
I don't understand why this is `Partial`. In `IosPlatformOptions` both `appleId` and `appleIdPassword` are never `undefined`:
```
appleId: string;
appleIdPassword: string;
```
`lodash.pick` makes this whole thing really complicated. Also not sure why the `credsPresent` checks below are necessary.

```suggestion
    const result = { appleId: this.options.appleId, appleIdPassword: this.options.appleIdPassword };
```

Reply:
My guess is that the code was originally written to be more generic than it currently needs to be, but I'd simplify it because it's just two fields we are interested in. No need for `pick` or `intersection`.

Reply:
I incorrectly assumed that appleId and appleIdPassword were always defined.